iroh/src/protocol.rs

```diff
@@ -100,6 +99,46 @@ pub struct RouterBuilder {
     protocols: ProtocolMap,
 }
 
+#[allow(missing_docs)]
+#[derive(Debug, Snafu)]
+pub enum ProtocolError {
+    #[snafu(transparent)]
+    Connect {
+        source: crate::endpoint::ConnectionError,
+    },
+    #[snafu(transparent)]
+    User {
+        source: Box<dyn std::error::Error + Send + Sync + 'static>,
+    },
+}
```

Comment:
@dignifiedquire questions:

The `User` variant is meant for when users need to write custom protocol errors in their protocol implementations, right? If so, thoughts on `Custom` rather than `User`.

We don't have `backtrace` or `span_trace` anywhere on here, even for the `Connect` variant. That makes sense to me for the `User`/`Custom` variant, but is there are reason it's missing from `Connect`?